catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveCatalogOperations.java

```diff
@@ -135,11 +152,87 @@ public void initialize(Map<String, String> conf) throws RuntimeException {
     // and gravitinoConfig will be passed to Hive config, and gravitinoConfig has higher priority
     mergeConfig.forEach(hadoopConf::set);
     hiveConf = new HiveConf(hadoopConf, HiveCatalogOperations.class);
+    UserGroupInformation.setConfiguration(hadoopConf);
+
+    initKerberosIfNecessary(conf, hadoopConf);
 
     this.clientPool =
         new CachedClientPool(getClientPoolSize(conf), hiveConf, getCacheEvictionInterval(conf));
   }
 
+  private void initKerberosIfNecessary(Map<String, String> conf, Configuration hadoopConf) {
+    if (UserGroupInformation.AuthenticationMethod.KERBEROS
+        == SecurityUtil.getAuthenticationMethod(hadoopConf)) {
+      try {
+        File keytabTemporaryDirectory = new File("tmp");
+        if (!keytabTemporaryDirectory.exists()) {
+          // Ignore the return value, because there exists many Hive catalog operations making
+          // this directory.
+          keytabTemporaryDirectory.mkdir();
+        }
+
+        // The id of entity is a random unique id.
+        File keytabFile = new File(String.format("tmp/gravitino-%s-keytab", entity.id()));
```

Comment:
It is a unique id, not a random id. So if you have your gravitino restart, the id will not changed, so the keytab will be existed anyway.

Reply:
The file is deleteOnExit.